packages/did/src/Did.chain.ts

```diff
@@ -403,14 +401,70 @@ export async function generateCreateTxFromDidDetails(
   const signature = await signer.sign({
     data: encodedDidCreationDetails.toU8a(),
     meta: {},
-    publicKey: Crypto.coToUInt8(signingPublicKey),
-    alg,
+    publicKey: Crypto.coToUInt8(authenticationKey.publicKey),
+    alg: getSignatureAlgForKeyType(authenticationKey.type) as string,
   })
   return api.tx.did.create(encodedDidCreationDetails, {
     [signature.alg]: signature.data,
   })
 }
 
+export async function generateCreateTxFromDidDetails(
```

Comment:
S: Not sure what this function is exactly for (assuming full did creation?). Could you please add a comment?

Reply:
https://github.com/KILTprotocol/sdk-js/pull/473/commits/9048f4ba1c9a09644913d2dada338b28187ef7d0